ingestion/justice_data_source/api_client.py

```diff
@@ -104,3 +104,11 @@ def _get_publication_metadata(
                 break
 
         return last_updated_timestamp, refresh_frequency, owner_email
+
+    def validate_domains(self, datahub_domains) -> bool:
```

Comment:
I think this could benefit from set operations to avoid checking the same value multiple times.

E.g. something like 

```python
for domain in set(self._id_to_domain_mapping.values()):
  if domain.lower() not in set(datahub_domains):
```